docs/code-examples/repository-pattern/01_simple_example/ilObjGeoLocation.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+class ilObjGeoLocation {
+
+    protected $id;
+    protected $title;
+    protected $lattitude;
+    protected $longitude;
+    protected $expiration_timestamp;
+
+    public function __construct($a_id, $a_title, $a_lattitude, $a_longitude, $a_expiration_timestamp)

```

Comment:
Please add typehints for the parameters. This is only tangentially related to the repository pattern, but we would like to push this mechanism in the context of the [input processing](https://github.com/ILIAS-eLearning/ILIAS/pull/1371).

Reply:
I would use `\DateTimeImmutable` or just `\DateTime` atm, but I did not ponder that much. Its rather what I would *not* use: I would not use int and float, since they rely on an unstable external definition of some zero-time and I would not use `ilDateTime` because I cannot see advantages over PHPs buildin types.